gensim/models/wrappers/ldamallet.py

```diff
@@ -305,14 +305,9 @@ def read_doctopics(self, fname, eps=1e-6, renorm=True):
                 # the MALLET doctopic format changed in 2.0.8 to exclude the id,
                 # this handles the file differently dependent on the pattern
                 if len(parts) == 2 * self.num_topics:
-                    doc = [(id_, weight)
-                           for id_, weight in zip(map(int, parts[::2]),
-                                                  map(float, parts[1::2]))
-                           if abs(weight) > eps]
+                    doc = [(id_, weight) for id_, weight in ((int(x), float(y)) for (x, y) in zip(parts[::2], parts[1::2])) if abs(weight) > eps]
                 elif len(parts) == self.num_topics and mallet_version != '2.0.7':
-                    doc = [(id_, weight)
-                           for id_, weight in enumerate(map(float, parts))
-                           if abs(weight) > eps]
+                    doc = [(id_, weight) for id_, weight in enumerate(float(x) for x in parts) if abs(weight) > eps]
```

Comment:
@menshikh-iv 
line 308:
```
doc = [(int(id_), float(weight))
           for id_, weight in zip(*[iter(parts)] * 2)
           if abs(float(weight)) > eps]
```

line 310 :
```
doc = [(id_, float(weight))
           for id_, weight in enumerate(parts)
           if abs(float(weight)) > eps]
```
How about above code?
Above code is less loop

Reply:
Ok, accepted.

Reply:
@zsef123 Both examples above are incorrect. The correct formatting using hanging indent:

```python
doc = [
    (id_, float(weight))
    for id_, weight in enumerate(parts)
    if abs(float(weight)) > eps]
```

or (I personally prefer this):

 ```python
doc = [
    (id_, float(weight))
    for id_, weight in enumerate(parts)
    if abs(float(weight)) > eps
]
```

Actually, this line not too long, so simple:

```python
doc = [(id_, float(weight)) for id_, weight in enumerate(parts) if abs(float(weight)) > eps]
```

works too.

Reply:
already done as your last variant @piskvorky